dotnet/samples/GettingStartedWithProcesses/Step04/Steps/ManagerAgentStep.cs

```diff
@@ -0,0 +1,146 @@
+    }
+
+    [KernelFunction(Functions.ReceiveResponse)]
+    public async Task ReceiveResponseAsync(KernelProcessStepContext context, Kernel kernel, string response)
+    {
+        // Get the chat history
+        IChatHistoryProvider historyProvider = kernel.GetHistory();
+        ChatHistory history = await historyProvider.GetHistoryAsync();
+
+        // Proxy the inner response
+        ChatCompletionAgent agent = kernel.GetAgent<ChatCompletionAgent>(AgentServiceKey);
+        ChatMessageContent message = new(AuthorRole.Assistant, response) { AuthorName = agent.Name };
+        history.Add(message);
+
+        await context.EmitEventAsync(new() { Id = AgentOrchestrationEvents.AgentResponse, Data = message });
+
+        await context.EmitEventAsync(new() { Id = AgentOrchestrationEvents.AgentResponded });
+    }
+
+    private static async Task<IntentResult> IsRequestingUserInputAsync(Kernel kernel, ChatHistory history, ILogger logger)
+    {
+        ChatHistory localHistory =
+        [
+            new ChatMessageContent(AuthorRole.System, "Analyze the conversation and determine if user input is being solicited."),
+            .. history.TakeLast(1)
+        ];
+
+        IChatCompletionService service = kernel.GetRequiredService<IChatCompletionService>();
+
+        ChatMessageContent response = await service.GetChatMessageContentAsync(localHistory, new OpenAIPromptExecutionSettings { ResponseFormat = s_intentResponseFormat });
+        IntentResult intent = JsonSerializer.Deserialize<IntentResult>(response.ToString())!;
+
+        logger.LogTrace("{StepName} Response Intent - {IsRequestingUserInput}: {Rationale}", nameof(ManagerAgentStep), intent.IsRequestingUserInput, intent.Rationale);
+
+        return intent;
+    }
+
+    private static readonly ChatResponseFormat s_intentResponseFormat = ChatResponseFormat.CreateJsonSchemaFormat(
+        jsonSchemaFormatName: "intent_result",
+        jsonSchema: BinaryData.FromString(
+        """
+        {
+            "type": "object",
+            "properties": {
+                "IsRequestingUserInput": {
+                    "type": "boolean",
+                    "description": "True if user input is requested or solicited.  Addressing the user with no specific request is False.  Asking a question to the user is True."
+                },
+                "IsWorking": {
+                    "type": "boolean",
+                    "description": "True if the user request is being worked on."
+                },
+                "Rationale": {
+                    "type": "string",
+                    "description": "Rationale for the value assigned to IsRequestingUserInput"
+                }
+            },
+            "required": ["IsRequestingUserInput", "IsWorking", "Rationale"],
+            "additionalProperties": false
+        }
+        """),
+        jsonSchemaIsStrict: true);
+
+    private sealed class IntentResult
```

Comment:
[not for this PR]: since there will be more agent + structured output samples that rely on json schemas perhaps we could start using in the samples libs like `JsonSchema.Net` that includes annotators so then the classes already include the schema definition.

Probably the IntentResult would look like:

``` c#
[Title("IntentResult")]
private sealed class IntentResult
{
        [Required]
        [Description("True if user input is requested or solicited.  Addressing the user with no specific request is False.  Asking a question to the user is True.")]
        public bool IsRequestingUserInput { get; set; }
        
        [Required]
        [Description("True if the user request is being worked on.")]
        public bool IsWorking { get; set; }
        
        [Required]
        [Description("Rationale for the value assigned to IsRequestingUserInput")]
        public string Rationale { get; set; }
}
```